src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -433,7 +433,7 @@ impl KvmVcpu {
         // per million beacuse it is common for TSC frequency
         // to differ due to calibration at boot time.
         let diff = (i64::from(self.get_tsc_khz()?) - i64::from(state_tsc_freq)).abs();
-        Ok(diff > i64::from(state_tsc_freq * TSC_KHZ_TOL_NUMERATOR / TSC_KHZ_TOL_DENOMINATOR))
+        Ok(diff > i64::from(state_tsc_freq) * TSC_KHZ_TOL_NUMERATOR / TSC_KHZ_TOL_DENOMINATOR)
```

Comment:
Mh, this kinda just punts the issue to instead happen at `300000000000GHz`. Maybe we can use `checked_mul` and propagate the error?

Reply:
does it? `state_tsc_freq` remains a `u32`. So the max value of it will be `u32::MAX`. I think that `i64::from(u32::MAX)` * 250 should not overflow, right?

That said, doing the `checked_mul` thing is probably a better way to handle this anyways.

Reply:
Good point, I will try and fix all those